test/utils/ansible-test/reorganize-tests.sh

```diff
@@ -0,0 +1,243 @@
+for type in test_win_group1 test_win_group2 test_win_group3; do
+    targets=$(grep 'role:' "test/integration/${type}.yml" \
+        | sed 's/^.* role: //; s/[ ,].*$//;')
+
+    group=$(echo "${type}" | sed 's/^test_win_/windows_/; s/_/\/ci\//;')
+
+    for target in ${targets}; do
+        echo "target: ${target}, group: ${group}"
+        echo "${group}" >> "test/integration/targets/${target}/aliases"
+    done
+done
+
+# Add additional windows tests to appropriate groups.
+echo 'windows/ci/group2' >> test/integration/targets/binary_modules_winrm/aliases
+echo 'windows/ci/group3' >> test/integration/targets/connection_winrm/aliases
+
+# Add posix/ci/group3 for posix tests which are not already grouped for ci.
+group="posix/ci/group3"
+for target in test/integration/targets/*; do
+    target=$(basename "${target}")
+    if [[ "${target}" =~ (setup|prepare)_ ]]; then
+        continue
+    fi
+    if [ -f "test/integration/targets/${target}/test.sh" ]; then
+        continue
+    fi
+    if [ -f "test/integration/targets/${target}/aliases" ]; then
+        if grep -q -P "^(windows|posix)/" "test/integration/targets/${target}/aliases"; then
+            continue
+        fi
+    fi
+    if [[ "${target}" =~ _ ]]; then
+        prefix="${target//_*/}"
+        if grep -q --line-regex "${prefix}" test/integration/target-prefixes.*; then
+            continue
+        fi
+    fi
+    echo "target: ${target}, group: ${group}"
+    echo "${group}" >> "test/integration/targets/${target}/aliases"
+done
+
+# Add skip aliases for python3.
+sed 's/^test_//' test/utils/shippable/python3-test-tag-blacklist.txt | while IFS= read -r target; do
+    echo "skip/python3" >> "test/integration/targets/${target}/aliases"
+done
+
+# Add skip aliases for tests which don't pass yet on osx/freebsd.
+for target in service postgresql mysql_db mysql_user mysql_variables uri get_url async_extra_data; do
+    echo "skip/osx" >> "test/integration/targets/${target}/aliases"
+    echo "skip/freebsd" >> "test/integration/targets/${target}/aliases"
+done
+
+# Add skip aliases for tests which don't pass yet on osx.
+for target in gathering_facts iterators git; do
+    echo "skip/osx" >> "test/integration/targets/${target}/aliases"
+done
+
+# Add needs/root entries as required.
+for target in connection_chroot authorized_key copy template unarchive; do
+    echo "needs/root" >> "test/integration/targets/${target}/aliases"
+done
+
+# Add missing alias for windows async_status.
+echo "async_status" >> test/integration/targets/win_async_wrapper/aliases
+
+# Remove connection tests from CI groups which aren't supported yet.
+for connection in docker jail libvirt_lxc lxc lxd; do
+    target="connection_${connection}"
+    sed -i '/^posix\/ci\/.*$/d' "test/integration/targets/${target}/aliases"
+done
+
+# FIXME: Remove tests from CI which are currently broken under ansible-test. Fix these before deploying ansible-test.
```

Comment:
I'm using the `FIXME` entries as the list to track these. All the `FIXME` entries should be resolved and removed before the PR is merged.